storage/storage-interface/src/finality_view.rs

```diff
@@ -0,0 +1,159 @@
+use std::sync::RwLock;
+
+use aptos_crypto::HashValue;
+use aptos_types::{
+    aggregate_signature,
+    block_info::BlockInfo,
+    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+    transaction::Version,
+};
+
+use crate::{AptosDbError, DbReader, Result};
+
+/// A wrapper over [`DbReader`], representing ledger at the end of a block
+/// as the latest ledger and its version as the latest transaction version.
+pub struct FinalityView<Db> {
```

Comment:
I feel we may want to call this `FinView` since we are referring specifically to the `fin` concept outlined here: https://github.com/movementlabsxyz/rfcs/tree/main/0018-m2-state-finality-modes

Reply:
I was naming things with my aptos-core hat on, so I used more general terms in this code base.